src/main/java/teammates/ui/newcontroller/GetCourseStudentDetailsAction.java

```diff
@@ -0,0 +1,101 @@
+package teammates.ui.newcontroller;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.datatransfer.attributes.StudentProfileAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.EntityNotFoundException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+
+/**
+ * Action: gets details of a student in a course.
+ */
+public class GetCourseStudentDetailsAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isInstructor) {
+            throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
+        }
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmail = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            throw new EntityNotFoundException(
+                new EntityDoesNotExistException("No student with given email in given course."));
+        }
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId), student.section,
+                                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS);
+    }
+
+    @Override
+    public ActionResult execute() {
+
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmail = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
+        if (student == null) {
+            return new JsonResult("No student with given email in given course.", HttpStatus.SC_NOT_FOUND);
+        }
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(instructor, logic.getCourse(courseId), student.section,
+                                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS);
+
+        boolean hasSection = false;
+        try {
+            hasSection = logic.hasIndicatedSections(courseId);
+        } catch (EntityDoesNotExistException e) {
+            return new JsonResult("No course with given id.", HttpStatus.SC_NOT_FOUND);
+        }
+
+        StudentProfileAttributes studentProfile = logic.getStudentProfile(student.googleId);
+
+        student.googleId = null;
+        student.key = null;
+        studentProfile.googleId = null;
```

Comment:
When testing, I got an NPE when `studentProfile` is null. This needs a non-null check. Also needs a test to prevent that from happening again.